packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -1118,12 +1118,32 @@ export const transformers = (model: ModelData): string[] => {
 
 	if (model.pipeline_tag && LIBRARY_TASK_MAPPING.transformers?.includes(model.pipeline_tag)) {
 		const pipelineSnippet = ["# Use a pipeline as a high-level helper", "from transformers import pipeline", ""];
+		let useMessagesSyntax = false;
 
-		if (model.tags.includes("conversational") && model.config?.tokenizer_config?.chat_template) {
-			pipelineSnippet.push("messages = [", '    {"role": "user", "content": "Who are you?"},', "]");
+		if (model.config?.tokenizer_config?.chat_template) {
+			if (model.tags.includes("image-text-to-text") && model.tags.includes("conversational")) {
+				pipelineSnippet.push(
+					"messages = [",
+					[
+						"    {",
+						'        "role": "user",',
+						'        "content": [',
+						'            {"type": "image", "url": "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/p-blog/candy.JPG"},',
+						'            {"type": "text", "text": "What animal is on the candy?"}',
+						"        ]",
+						"    },",
+					].join("\n"),
+					"]"
+				);
+				useMessagesSyntax = true;
+			} else if (model.tags.includes("conversational")) {
```

Comment:
Don't we need `messages` here as well?

Reply:
Perhaps we can do the flow like `if conversational and chat_template`, then check whether image-text-to-text or not, and remove the `useMessagesSyntax` var.

Reply:
good point